modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/LockHolder.java

```diff
@@ -36,7 +35,7 @@
 class LockHolder<T extends Lock> {
     private final T lock;
 
-    private final AtomicInteger lockHolder = new AtomicInteger();
+    private int lockHoldersCount;
```

Comment:
"compute" guarantees mutual exclusivity of closures and HB between everything, as far as I know. I'll add a comment

Reply:
To guarantee this, `compute()` probably does volatile writes, maybe with some `synchronized`. So making the field `volatile` should not cost us anything (everything is already paid), but it will make the code less assumptive (so after a change, it will be more difficult to break it)

Reply:
But why would I make something "volatile" or atomic if it doesn't need to be? This is misleading